tbdex/rfq/rfq.go

```diff
@@ -166,6 +173,213 @@ func (r *RFQ) verifyPrivateData() error {
+	paymentMethods := make([]offering.PaymentMethod, len(payoutMethods))
+	for i, method := range payoutMethods {
+		paymentMethods[i] = method
+	}
+	return paymentMethods
+}
+
+func (r *RFQ) verifyPaymentMethod(
+	selectedPaymentKind string,
+	selectedPaymentDetailsHash string,
+	privateData *PrivateData,
+	offeringPaymentMethods []offering.PaymentMethod,
+	payDirection PayDirection,
+) error {
+	// Filter allowed payment methods by selectedPaymentKind
+	var allowedPaymentMethods []offering.PaymentMethod
+	for _, paymentMethod := range offeringPaymentMethods {
+		if paymentMethod.GetKind() == selectedPaymentKind {
+			allowedPaymentMethods = append(allowedPaymentMethods, paymentMethod)
+		}
+	}
+
+	// If no matching payment methods are found, throw an error
+	if len(allowedPaymentMethods) == 0 {
+		var allowedPaymentMethodKinds []string
+		for _, paymentMethod := range offeringPaymentMethods {
+			allowedPaymentMethodKinds = append(allowedPaymentMethodKinds, paymentMethod.GetKind())
+		}
+		return fmt.Errorf(
+			"offering does not support rfq's %sMethod kind. (rfq) %s was not found in: [%s] (offering)",
+			payDirection, selectedPaymentKind, strings.Join(allowedPaymentMethodKinds, ", "),
+		)
+	}
+
+	for _, paymentMethod := range allowedPaymentMethods {
+		if paymentMethod.GetRequiredPaymentDetails() == nil {
+			// If requiredPaymentDetails is omitted, and selectedPaymentDetailsHash is also omitted, we have a match
+			if selectedPaymentDetailsHash == "" {
+				return nil
+			}
+			// selectedPaymentDetailsHash is present even though requiredPaymentDetails is omitted.
+			return errors.New("rfq paymentDetails must be omitted when offering requiredPaymentDetails is omitted")
+		}
+
+		// todo do we actually want to return error in this case where requiredPaymentDetails is present but privateData is nil?
+		if privateData == nil {
+			return errors.New("offering requiredPaymentDetails is present but rfq private data is omitted")
+		}
+
+		// requiredPaymentDetails is present and privateData is present, so Rfq's payment details must match
+		if payDirection == PayIn {
+			err := validatePaymentDetailSchema(paymentMethod, privateData.Payin.PaymentDetails, payDirection)
+			if err != nil {
+				return err
+			}
+		}
+
+		if payDirection == PayOut {
+			err := validatePaymentDetailSchema(paymentMethod, privateData.Payout.PaymentDetails, payDirection)
+			if err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func validatePaymentDetailSchema(paymentMethod offering.PaymentMethod, paymentDetails map[string]any, payDirection PayDirection) error {
+	if paymentDetails == nil {
+		return fmt.Errorf("rfq %s paymentDetails are missing", payDirection)
+	}
+
+	sch, err := json.Marshal(paymentMethod.GetRequiredPaymentDetails())
+	if err != nil {
+		return fmt.Errorf("failed to marshal requiredPaymentDetails: %w", err)
+	}
+
+	// todo idk what the id should be?
+	schema, err := jsonschema.CompileString("schema.json", string(sch))
```

Comment:
i am passing in `"schema.json"` for the first argument `url string` but no idea if that makes sense